### DISCUSSION TITLE
rocketmq5.2.0版本，搭建主从发送消息 出现 Task was cancelled

### DISCUSSION THREAD

**Discussion by jaylu163** [apache/rocketmq#8722] [Q&A]
部署环境：

![image](https://github.com/user-attachments/assets/e2505e11-cf9c-439f-90f5-b6a457d60cad)
2个namesrv，1主1从的broker

![image](https://github.com/user-attachments/assets/fab3d9d9-4f70-4c8e-8bb8-aef311a3e12d)

![image](https://github.com/user-attachments/assets/f8ec8934-c58a-4ed9-a5ff-66609586f0ee)

各自broker的日志如下：
主发消息有一个连接日志
![image](https://github.com/user-attachments/assets/d831db00-6656-47af-82b6-f76f4b0f94a2)

从发消息日志没有请求，但是broker日志没有报错
![image](https://github.com/user-attachments/assets/7955aba6-1e5e-4ee8-b592-0a4755cef3bb)

客户端日志：

![image](https://github.com/user-attachments/assets/468066d8-9124-4a04-b9cf-6b09d49b7155)

不确定是我配置问题，还是5.2开启proxy后的bug

**Comment by luozongle01** [comment]
老哥  你这个demo第46行是啥啊，然后完整的异常堆栈发下看看呢

**Reply by jaylu163** [reply]
![image](https://github.com/user-attachments/assets/a98fe17b-a173-4cba-ab28-4bc66a6955e6)
这个代码是完整的示例

**Reply by jaylu163** [reply]
同样的代码，像单实例发送消息。是可以成功

**Reply by luozongle01** [reply]
> 相同的代码，就像单个实例发送消息。是可以成功的

3.174是个从节点吗，会不会是因为这里不能填从节点的地址啊。

**Reply by jaylu163** [reply]
没有看到官方文档说不可以填写，我也试过。去掉3.174这个从节点。还是报同样的错呢
很不解是哪块导致

**Reply by YongXiaoo** [reply]
我一直是分开搭建的，主从模式grpc有bug，可以直接单主试试，还可以设置下grpc客户端的超时时间

**Reply by luozongle01** [reply]
> ![image](https://private-user-images.githubusercontent.com/11609127/369268027-a98fe17b-a173-4cba-ab28-4bc66a6955e6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyNjU2NzIsIm5iZiI6MTczMDI2NTM3MiwicGF0aCI6Ii8xMTYwOTEyNy8zNjkyNjgwMjctYTk4ZmUxN2ItYTE3My00Y2JhLWFiMjgtNGJjNjZhNjk1NWU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDMwVDA1MTYxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5MzJmODAxZGI4ZWEwOTkwMDlkMzBhOGJkOTM0NDJiYjExNmQ4ZTJhM2VlZmEzMTZjY2RjMzc0MDU2MDcwNjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.B62G_aOGOtVaaWbRDYVx5Lo5Ux3B1VAzeMcaZ76rqQM) 这个代码是完整的示例

为什么3.174这个端口是8900呢，默认不应该是8088吗

**Reply by jaylu163** [reply]
local模式感觉是有bug，也不知道哪块问题

3.174的端口是被我重新定义设置了。8080，8081我有其他服务占用

**Reply by luozongle01** [reply]
> local模式感觉是有bug，也不知道哪块问题
> 
> 3.174的端口是被我重新定义设置了。8080，8081我有其他服务占用

local模式的情况下，proxy的grpcPort应该是不能更改，改了会有问题。

**Reply by jaylu163** [reply]
我验证了下，不修改，发消息也有问题。比较奇怪的。搞不懂这个是哪块问题
